GVRf/Framework/src/org/gearvrf/asynchronous/AsyncCubemapTexture.java

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.asynchronous;
+
+import java.io.IOException;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.gearvrf.GVRAndroidResource;
+import org.gearvrf.GVRAndroidResource.CancelableCallback;
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRCubemapTexture;
+import org.gearvrf.GVRHybridObject;
+import org.gearvrf.GVRTexture;
+import org.gearvrf.asynchronous.Throttler.AsyncLoader;
+import org.gearvrf.asynchronous.Throttler.AsyncLoaderFactory;
+import org.gearvrf.asynchronous.Throttler.GlConverter;
+import org.gearvrf.utility.Log;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+
+/**
+ * Async resource loading: cube map textures.
+ * 
+ * Since ZipInputStream does not support mark() and reset(), we directly use
+ * BitmapFactory .decodeStream() in loadResource().
+ * 
+ * @since 1.6.9
+ */
+abstract class AsyncCubemapTexture {
+
+    /*
+     * The API
+     */
+
+    static void loadTexture(GVRContext gvrContext,
+            CancelableCallback<GVRTexture> callback,
+            GVRAndroidResource resource, int priority) {
+        Throttler.registerCallback(gvrContext, TEXTURE_CLASS, callback,
+                resource, priority);
+    }
+
+    /*
+     * Static constants
+     */
+
+    private static final String TAG = Log.tag(AsyncCubemapTexture.class);
+
+    private static final Class<? extends GVRHybridObject> TEXTURE_CLASS = GVRTexture.class;
```

Comment:
I'm pretty sure this breaks asynch textures - you need to register `AsyncCubemapTexture` here.